src/main/scala/nl/knaw/dans/lib/error/package.scala

```diff
@@ -76,4 +76,145 @@ package object error {
         Success(xs.map(_.get).to(canBuildFrom))
     }
   }
+
+  implicit class FailFastStream[T](val stream: Stream[Try[T]]) {
+    /**
+     * Evaluates a `Stream` of `Try`s into either:
+     *  - one `Success` with a stream of `T`s or
+     *  - a `Failure` with the first exception contained in the stream
+     *
+     * Note that when the `Stream` encounters a `Failure`, the remaining elements are not evaluated.
+     * The `Failure` is returned immediately.
+     *
+     * @example
+     * {{{
+     *   import nl.knaw.dans.lib.error._
+     *
+     *   import scala.util.Try
+     *
+     *   def f(i: Int) = {
+     *     if (i <= 2) i
+     *     else throw new Exception(s"$$i is larger than 2")
+     *   }
+     *
+     *   val stream: Try[Stream[Int]] = (0 to 5).toStream.map(i => Try(f(i))).failFast
+     *   println(stream)
+     *   // prints: Failure(java.lang.Exception: 3 is larger than 2)
+     * }}}
+     *
+     * @return a evaluated result
+     */
+    def failFast: Try[Stream[T]] = {
+      stream.find(_.isFailure)
+        .map(_.flatMap(s => Failure(new IllegalArgumentException(s"Success should never occur here, but got Success($s)"))))
+        .getOrElse(Success(stream.map(_.get)))
+    }
+  }
+
+  implicit class TryExtensions[T](val t: Try[T]) extends AnyVal {
+    /**
+     * Applies the given side effecting function if and only if this is a `Success`.
+     *
+     * Example:
+     * {{{
+     *   import nl.knaw.dans.lib.error.TryExtensions
+     *
+     *   import scala.util.{Failure, Success, Try}
+     *
+     *   def getFileLength(file: File): Try[Long] =
+     *     if (file.exists) Success(file.length)
+     *     else Failure(new FileNotFoundException())
+     *
+     *   def performSideEffect(size: Long): Unit = println(s"size = $$size")
```

Comment:
otherwise the mvn scaladoc plugin is going to complain that it views `$size` as a variable. Apparently we escape a `$` with another `$`.